lib/util/eyeglass_modules.js

```diff
@@ -0,0 +1,283 @@
+  }, {});
+}
+
+function getModuleName(pkg) {
+  var name = normalizeEyeglassOptions(pkg.data.eyeglass).name || pkg.data.name;
+  if (!name) {
+    throw new Error(tick(pkg.path) + " does not contain a valid module name");
+  }
+  return name;
+}
+
+function normalizeEyeglassOptions(options) {
+  if (typeof options === "string") {
+    return {
+      exports: options
+    };
+  }
+  return typeof options === "object" ? options : {};
+}
+
+function getFinalModule(name) {
+  return this.collection[name];
+}
+
+function getExportsFileFromOptions(options) {
+  return normalizeEyeglassOptions(options).exports;
+}
+
+function getModuleExports(pkg, modulePath) {
+  var exportsFile = getExportsFileFromOptions(pkg.eyeglass) || pkg.main;
+  return path.join(modulePath, exportsFile || "");
+}
+
+function isEyeglassModule(pkg) {
+  return !!(pkg && includes(pkg.keywords, EYEGLASS_KEYWORD));
+}
+
+function resolveModule(pkg, isRoot) {
+  pkg = packageUtils.getPackage(pkg);
+  var isEyeglassMod = isEyeglassModule(pkg.data);
+  if (isEyeglassMod || (pkg.data && isRoot)) {
+    var modulePath = path.dirname(pkg.path);
+    if (typeof pkg.data.version === "object") {
+      console.log("pkg.data.version", pkg.data.version);
+    }
+    return {
+      name: getModuleName(pkg) || (isRoot && ROOT_ID),
+      rawName: pkg.data.name,
+      path: modulePath,
+      version: pkg.data.version,
+      main: getModuleExports(pkg.data, modulePath),
+      isEyeglassModule: isEyeglassMod,
+      dependencies: discoverModules({
+        dir: modulePath,
+        isRoot: isRoot
+      }),
+      eyeglass: normalizeEyeglassOptions(pkg.data.eyeglass)
+    };
+  }
+}
+
+function getEyeglassSelf(options) {
+  var eyeglassDir = path.resolve(__dirname, "..", "..");
+  var eyeglassPkg = packageUtils.getPackage(eyeglassDir);
+
+  var resolvedPkg = resolve(eyeglassPkg.path, eyeglassPkg.path, eyeglassDir);
+  return resolveModule(resolvedPkg);
+}
+
+function discoverModules(options) {
+  var pkg = options.pkg || packageUtils.getPackage(options.dir);
+
+  var dependencies = {};
+
+  // get the collection of dependencies
+  // merge the devDependencies only if it's the root of the project
+  if (pkg.data) {
+    dependencies = merge(
+      dependencies,
+      pkg.data.dependencies,
+      options.isRoot && pkg.data.devDependencies
+    );
+  }
+
+  dependencies = Object.keys(dependencies).reduce(function(modules, dependency) {
+    var resolvedPkg = resolve(packageUtils.getPackagePath(dependency), pkg.path, options.dir);
+    var resolvedModule = resolveModule(resolvedPkg);
+    if (resolvedModule) {
+      modules[resolvedModule.name] = resolvedModule;
+    }
+    return modules;
+  }, {});
+
+  // if it's the root...
+  if (options.isRoot) {
+    // ensure eyeglass itself is a direct dependency
+    dependencies.eyeglass = dependencies.eyeglass || getEyeglassSelf(options);
+  }
+
+
+  return Object.keys(dependencies).length ? dependencies : null;
+}
+
+function tick(str) {
```

Comment:
Is this the only place we're bothering with backticks around path stuff or are we missing a util file as part of the refactor?